datasets/bbc_hindi_nli/bbc_hindi_nli.py

```diff
@@ -0,0 +1,157 @@
+SOFTWARE.
+"""
+
+_TRAIN_DOWNLOAD_URL = (
+    "https://raw.githubusercontent.com/avinsit123/hindi-nli-data/master/Textual_Entailment/BBC/BBC_recasted_train.tsv"
+)
+_VALID_DOWNLOAD_URL = (
+    "https://raw.githubusercontent.com/avinsit123/hindi-nli-data/master/Textual_Entailment/BBC/BBC_recasted_dev.tsv"
+)
+_TEST_DOWNLOAD_URL = (
+    "https://raw.githubusercontent.com/avinsit123/hindi-nli-data/master/Textual_Entailment/BBC/BBC_recasted_test.tsv"
+)
+
+
+class BbcHindiNLIConfig(datasets.BuilderConfig):
+    """BuilderConfig for BBC Hindi NLI Config"""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for BBC Hindi NLI Config.
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(BbcHindiNLIConfig, self).__init__(**kwargs)
+
+
+class BbcHindiNLI(datasets.GeneratorBasedBuilder):
+    """BBC Hindi NLI dataset -- Dataset providing textual-entailment pairs for NLI tasks in Hindi"""
+
+    BUILDER_CONFIGS = [
+        BbcHindiNLIConfig(
+            name="bbc hindi nli",
+            version=datasets.Version("1.1.0"),
+            description="BBC Hindi NLI: Natural Language Inference Dataset in Hindi",
+        ),
+    ]
+
+    def _info(self):
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "premise": datasets.Value("string"),
+                    "hypothesis": datasets.Value("string"),
+                    "label": datasets.Value("string"),
+                    "topic": datasets.ClassLabel(names=["india", "news" , "international" , "sport" , "science"]),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        train_path = dl_manager.download_and_extract(_TRAIN_DOWNLOAD_URL)
+        test_path = dl_manager.download_and_extract(_TEST_DOWNLOAD_URL)
+        valid_path = dl_manager.download_and_extract(_VALID_DOWNLOAD_URL)
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": train_path}),
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs={"filepath": valid_path}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": test_path}),
+        ]
+
+    def _generate_examples(self, filepath):
+        """ Yields examples. """
+
+        with open(filepath, encoding="utf-8") as tsv_file:
+            tsv_reader = csv.reader(tsv_file, delimiter="\t")
+            for id_, row in enumerate(tsv_reader):
+                (premise, hypothesis, label, topic) = row
+                yield id_, {
+                    "premise": premise,
+                    "hypothesis": hypothesis,
+                    "label": label,
+                    "topic": topic,
```

Comment:
```suggestion
                    "label": 1 if label == "entailed" else 0,
                    "topic": int(topic),
```
For class labels we expect to have integers here

Reply:
Done. pls check